packages/core/src/types/index.ts

```diff
@@ -1,10 +1,19 @@
-import { Camera, Frustum } from '@babylonjs/core';
-import { Tile, PriorityQueue } from '@tiledb-inc/viz-common';
+import { BoundingInfo } from '@babylonjs/core';
+import { Matrix } from 'mathjs';
+
+export type TDBNonEmptyDomain = {
+  isEmpty: boolean;
+  nonEmptyDomain: Record<string, number[]>;
+};
 
 export interface AssetMetadata {
-  dataset_type: string;
+  dataset_type?: string;
 }
 
+export type ImageAssetMetadata = AssetMetadata & {
+  metadata: string;
```

Comment:
I don't doubt this is correct, but, it's surprising to me -- I think of metadata usually as a map from string to string, or string to Any ... I'll have to read more to understand how this can be a single string ...

Reply:
`ImageAssetMetadata` describes the metadata schema of a image TileDB Array. The array metadata also happen to have a `metadata` field which stores the JSON representation of the image metadata with are parsed later back to an object